runtime/lua/vim/lsp/handlers.lua

```diff
@@ -215,9 +215,14 @@ M['textDocument/documentSymbol'] = symbol_handler
 M['workspace/symbol'] = symbol_handler
 
 --@see https://microsoft.github.io/language-server-protocol/specifications/specification-current/#textDocument_rename
-M['textDocument/rename'] = function(_, _, result)
-  if not result then return end
-  util.apply_workspace_edit(result)
+M['textDocument/rename'] = function(_, _, result, client_id)
+  if not result or not util.workspace_edit_has_edits(result) then
+    local client = vim.lsp.get_client_by_id(client_id)
+    local client_name = client and client.name or string.format("id=%d", client_id)
+    err_message("LSP[", client_name, "] Unable to rename symbol")
```

Comment:
Do you think it would make sense to include the `err` (first argument to the handler) here if it is non empty and not nil? Might contain additional information why the rename didn't work.

Reply:
So my thinking in not including it was that I'm trying to catch the situation where the symbol that is trying to be renamed, just can't be. So there isn't something that necessarily went wrong, but more that there is simply no document changes that are returned. I'd assume in the situation where something blows up and there is an `err` returned, that might not be best to show that to user as it's something that went wrong server side and should hopefully be in the logs. wdyt?

Reply:
> So there isn't something that necessarily went wrong, but more that there is simply no document changes that are returned.

I'd assume that if that is the case `err` would be `nil` anyhow, no? Depends on how language servers behave, but even if they include some message even if it is a regular "there is just nothing to rename" I think it wouldn't hurt exposing that mesage to the user.

No strong opinion on this.

Reply:
Yea, that makes sense. I went ahead and added that in.